### PR TITLE
use sqlalchemy 1.3.x for mautrix-telegram

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

The author seems to have no intentions of upgrading (source: https://github.com/tulir/mautrix-python/issues/46).
If they remove the feature that needs sqlalchemy, we can remove the 1.3.x version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127486` at 06e3c6b8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages marked as broken and skipped:</summary>
  <ul>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>gnuradio3_7Packages.rds</li>
    <li>hadoop</li>
    <li>hadoop_2_7</li>
    <li>hadoop_2_8</li>
    <li>hadoop_2_9</li>
    <li>impressive</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>natron</li>
    <li>ocropus</li>
    <li>odpdown</li>
    <li>opentsdb</li>
    <li>pig</li>
    <li>rainbowstream</li>
    <li>spark</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/b3ffd8e0264f4297fdd245b6deac05b8">python38Packages.pymatgen-lammps</a></li>
  </ul>
</details>
<details>
  <summary>1283 packages skipped due to time constraints:</summary>
  <ul>
    <li>DisnixWebService</li>
    <li>R</li>
    <li>abcl</li>
    <li>adapta-gtk-theme</li>
    <li>adoptopenjdk-icedtea-web</li>
    <li>aiodnsbrute</li>
    <li>alloy (alloy5)</li>
    <li>alloy4</li>
    <li>amidst</li>
    <li>ansible (ansible_2_11 ,python38Packages.ansible-core)</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>127 packages built successfully:</summary>
  <ul>
    <li>acd-cli</li>
    <li>anki</li>
    <li>ankisyncd</li>
    <li>clevercsv (python38Packages.clevercsv)</li>
    <li>cum</li>
    <li>graph-cli</li>
    <li>jre8</li>
    <li>liberation_ttf (liberation_ttf_v2)</li>
    <li>mailman</li>
    <li>mautrix-telegram</li>
    <li>python38Packages.APScheduler</li>
    <li>python38Packages.ColanderAlchemy</li>
    <li>python38Packages.Quandl (python38Packages.quandl)</li>
    <li>python38Packages.aiocontextvars</li>
    <li>python38Packages.aioinflux</li>
    <li>python38Packages.alembic</li>
    <li>python38Packages.apptools</li>
    <li>python38Packages.azure-core</li>
    <li>python38Packages.azure-storage-blob</li>
    <li>python38Packages.bkcharts</li>
    <li>python38Packages.celery</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.databases</li>
    <li>python38Packages.factory_boy</li>
    <li>python38Packages.flametree</li>
    <li>python38Packages.flammkuchen</li>
    <li>python38Packages.flask_sqlalchemy</li>
    <li>python38Packages.fonttools</li>
    <li>python38Packages.geoalchemy2</li>
    <li>python38Packages.google-api-core</li>
    <li>python38Packages.google-api-python-client</li>
    <li>python38Packages.google-auth</li>
    <li>python38Packages.google-auth-httplib2</li>
    <li>python38Packages.google-auth-oauthlib</li>
    <li>python38Packages.google-cloud-core</li>
    <li>python38Packages.google-cloud-iam</li>
    <li>python38Packages.google-cloud-kms</li>
    <li>python38Packages.google-cloud-storage</li>
    <li>python38Packages.google-cloud-testutils</li>
    <li>python38Packages.google-resumable-media</li>
    <li>python38Packages.gtts</li>
    <li>python38Packages.httplib2</li>
    <li>python38Packages.httpretty</li>
    <li>python38Packages.huggingface-hub</li>
    <li>python38Packages.influxdb</li>
    <li>python38Packages.kombu</li>
    <li>python38Packages.mailman</li>
    <li>python38Packages.mailman-hyperkitty</li>
    <li>python38Packages.marshmallow-sqlalchemy</li>
    <li>python38Packages.mautrix (python38Packages.mautrix-appservice)</li>
    <li>python38Packages.monty</li>
    <li>python38Packages.msrest</li>
    <li>python38Packages.oauth2client</li>
    <li>python38Packages.orm</li>
    <li>python38Packages.pandas</li>
    <li>python38Packages.partd</li>
    <li>python38Packages.pecan</li>
    <li>python38Packages.phik</li>
    <li>python38Packages.pyathena</li>
    <li>python38Packages.pygtfs</li>
    <li>python38Packages.pymatgen</li>
    <li>python38Packages.pymysqlsa</li>
    <li>python38Packages.pytest-randomly</li>
    <li>python38Packages.pytest-regressions</li>
    <li>python38Packages.python-telegram-bot</li>
    <li>python38Packages.pytrends</li>
    <li>python38Packages.seaborn</li>
    <li>python38Packages.sqlalchemy (python38Packages.sqlalchemy_1_3)</li>
    <li>python38Packages.sqlalchemy-citext</li>
    <li>python38Packages.sqlalchemy-i18n</li>
    <li>python38Packages.sqlalchemy-utils</li>
    <li>python38Packages.sqlalchemy_1_4</li>
    <li>python38Packages.sqlsoup</li>
    <li>python38Packages.starlette</li>
    <li>python38Packages.telethon-session-sqlalchemy</li>
    <li>python38Packages.tld</li>
    <li>python38Packages.tqdm</li>
    <li>twine (python38Packages.twine)</li>
    <li>python38Packages.vega_datasets</li>
    <li>python38Packages.vidstab</li>
    <li>python38Packages.yfinance</li>
    <li>python39Packages.ColanderAlchemy</li>
    <li>python39Packages.aiocontextvars</li>
    <li>python39Packages.alembic</li>
    <li>python39Packages.azure-core</li>
    <li>python39Packages.azure-storage-blob</li>
    <li>python39Packages.databases</li>
    <li>python39Packages.factory_boy</li>
    <li>python39Packages.flask_sqlalchemy</li>
    <li>python39Packages.geoalchemy2</li>
    <li>python39Packages.google-api-core</li>
    <li>python39Packages.google-api-python-client</li>
    <li>python39Packages.google-auth</li>
    <li>python39Packages.google-auth-httplib2</li>
    <li>python39Packages.google-auth-oauthlib</li>
    <li>python39Packages.google-cloud-core</li>
    <li>python39Packages.google-cloud-iam</li>
    <li>python39Packages.google-cloud-kms</li>
    <li>python39Packages.google-cloud-testutils</li>
    <li>python39Packages.google-resumable-media</li>
    <li>python39Packages.httplib2</li>
    <li>python39Packages.httpretty</li>
    <li>python39Packages.kombu</li>
    <li>python39Packages.msrest</li>
    <li>python39Packages.oauth2client</li>
    <li>python39Packages.pymysqlsa</li>
    <li>python39Packages.pytest-randomly</li>
    <li>python39Packages.sqlalchemy (python39Packages.sqlalchemy_1_3)</li>
    <li>python39Packages.sqlalchemy-citext</li>
    <li>python39Packages.sqlalchemy-i18n</li>
    <li>python39Packages.sqlalchemy-utils</li>
    <li>python39Packages.sqlalchemy_1_4</li>
    <li>python39Packages.sqlsoup</li>
    <li>python39Packages.starlette</li>
    <li>python39Packages.telethon-session-sqlalchemy</li>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.coresrht</li>
    <li>sourcehut.dispatchsrht</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.todosrht</li>
    <li>truvari</li>
    <li>vorta</li>
    <li>yokadi</li>
  </ul>
</details>
<details>
  <summary>7 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/servers/mautrix-telegram/default.nix:74:0:
```
   |
74 |     description = "A Matrix-Telegram hybrid puppeting/relaybot bridge";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/mautrix-telegram/default.nix:24:15:
```
   |
24 |   patches = [ ./0001-Re-add-entrypoint.patch ./0002-Don-t-depend-on-pytest-runner.patch ];
   |               ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pytest_xdist`.
Near pkgs/development/python-modules/sqlalchemy/1.3.nix:5:3:
```
  |
5 | , pytest_xdist
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/mautrix-telegram/default.nix:24:46:
```
   |
24 |   patches = [ ./0001-Re-add-entrypoint.patch ./0002-Don-t-depend-on-pytest-runner.patch ];
   |                                              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/sqlalchemy/1.4.nix:47:3:
```
   |
47 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/sqlalchemy/1.3.nix:40:3:
```
   |
40 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pytest_xdist`.
Near pkgs/development/python-modules/sqlalchemy/1.4.nix:12:3:
```
   |
12 | , pytest_xdist
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by petabyteboy** [comment]
This was not supposed to change the default version. Fixed that.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8121,7 +8121,9 @@ in {
 
   spyder-kernels_0_5 = callPackage ../development/python-modules/spyder-kernels/0.x.nix { };
 
-  sqlalchemy = callPackage ../development/python-modules/sqlalchemy { };
+  sqlalchemy_1_3 = callPackage ../development/python-modules/sqlalchemy/1.3.nix { };
+  sqlalchemy_1_4 = callPackage ../development/python-modules/sqlalchemy/1.4.nix { };
+  sqlalchemy = self.sqlalchemy_1_4;
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:8126]:
```suggestion
  sqlalchemy = callPackage ../development/python-modules/sqlalchemy { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/top-level/python-packages.nix:8126]: `sqlalchemy/default.nix` doesn't exist anymore, so this would break evaluation.

**Review by Ma27 — APPROVED** [review]

**Pull request merged**